File: doc/codes/python/examples/api/compute/modulation/sine.py
from autd3.units import Hz, rad
from autd3.value import Nearest, SamplingConfig
from autd3_modulation import SineOption, modulation_buffer, sine

freq = 150.0 * Hz
option = (
    # ANCHOR: option
    SineOption(
        amplitude=0xFF,
        offset=0x80,
        phase=0.0 * rad,
        clamp=False,
        sampling_config=SamplingConfig.FREQ_4K,
    )
    # ANCHOR_END: option
)
out = modulation_buffer()
# ANCHOR: api
sine(freq, option, out)
# ANCHOR_END: api

# ANCHOR: nearest
sine(Nearest(150.5 * Hz), option, out)
# ANCHOR_END: nearest

File: doc/codes/csharp/examples/api/compute/modulation/sine.cs
using AUTD3;
using static AUTD3.Units;

namespace DocSamples.ApiComputeModulationSine;

internal static class Sample
{
    internal static void Run()
    {
        var freq = 150 * Hz;
        byte amplitude = 0xFF;
        byte offset = 0x80;
        var phase = 0.0f * rad;
        var clamp = false;
        var samplingConfig = SamplingConfig.Freq4k;
        var option =
            // ANCHOR: option
            new SineOption(
                amplitude,
                offset,
                phase,
                clamp,
                samplingConfig
            )
            // ANCHOR_END: option
            ;
        var dst = Modulation.ModulationBuffer();
        // ANCHOR: api
        Modulation.Sine(freq, option, dst);
        // ANCHOR_END: api

        // ANCHOR: nearest
        Modulation.Sine(Nearest(150.5f * Hz), option, dst);
        // ANCHOR_END: nearest
    }
}
